x/configurl/oob.go

```diff
@@ -0,0 +1,48 @@
+package configurl
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/x/oob"
+)
+
+func registerOOBStreamDialer(r TypeRegistry[transport.StreamDialer], typeID string, newSD BuildFunc[transport.StreamDialer]) {
+	r.RegisterType(typeID, func(ctx context.Context, config *Config) (transport.StreamDialer, error) {
+		sd, err := newSD(ctx, config.BaseConfig)
+		if err != nil {
+			return nil, err
+		}
+		params := config.URL.Opaque
+
+		splitStr := strings.Split(params, ":")
```

Comment:
nit: maybe named arguments? https://github.com/Jigsaw-Code/outline-sdk/pull/324/files#diff-974d5fac785c258e952052dd1f6aa5b012bf2e4e43eb708de99420175a67e99b

But it's a matter of preference. @fortuna should know better.

Also, maybe you can make most arguments optional? I believe most users can use default timeout, default URG byte value, etc.